### PR TITLE
Change binding on Hide REPL to M-,

### LINKED ISSUE #1325: Remap dialog collapse key to ALT-comma
Remap dialog collapse key to CTRL-comma.

Currently there is a conflict with CTRL-s used for "saving" the map in the world editor (#873).

_Originally posted by @kostmo in https://github.com/swarm-game/swarm/discussions/1312#discussioncomment-6135624_

### PR DESCRIPTION
**Pull request by noahyor**
Fixes #1325.

### CODE REVIEW

File: src/Swarm/TUI/Controller.hs [hunk 1]
```diff
@@ -387,7 +387,7 @@ handleMainEvent ev = do
       -- Paint with the World Editor
       EC.handleCtrlLeftClick mouseLoc
     -- toggle collapse/expand REPL
-    ControlChar 's' -> do
+    ControlChar ',' -> do
```
Comment (kostmo) [inline src/Swarm/TUI/Controller.hs:390]: I just realized that, unfortunately, `CTRL + comma` is [not handled properly by the terminal](https://stackoverflow.com/a/7544277/105137).

However `ALT + comma` does work.  So we could change this to 
```suggestion
    MetaChar ',' -> do
```

**Review by kostmo — CHANGES_REQUESTED** [review]

File: src/Swarm/TUI/View.hs [hunk 2]
```diff
@@ -929,7 +929,7 @@ drawKeyMenu s =
       , may isPaused (NoHighlight, "^o", "step")
       , may (isPaused && hasDebug) (if s ^. uiState . uiShowDebug then Alert else NoHighlight, "M-d", "debug")
       , Just (NoHighlight, "^zx", "speed")
-      , Just (NoHighlight, "^s", if s ^. uiState . uiShowREPL then "hide REPL" else "show REPL")
+      , Just (NoHighlight, "^,", if s ^. uiState . uiShowREPL then "hide REPL" else "show REPL")
```
Comment (kostmo) [inline src/Swarm/TUI/View.hs:932]: The `Alt` key is the same as `Meta`, so this should be:
```suggestion
      , Just (NoHighlight, "M-,", if s ^. uiState . uiShowREPL then "hide REPL" else "show REPL")
```

**Comment by kostmo** [comment]
Thank you for pushing this one along!

**Review by kostmo — APPROVED** [review]

**Review by byorgey — APPROVED** [review]
I tested the new `M-,` , seems to work.

**Pull request merged**